Source/cmMakefileTargetGenerator.cxx

```diff
@@ -189,9 +189,13 @@ void cmMakefileTargetGenerator::CreateLinkCommandFile(
   const std::vector<std::string>& makefile_depends,
   const std::set<int>& ranlibIndexes)
 {
-  if (this->Makefile->IsOn("CMAKE_EXPORT_LINK_COMMANDS")) {
-    cmMakefileTargetGenerator::CreateLinkScriptJSON(
-      link_commands, makefile_depends, ranlibIndexes);
+  std::string version =
+    cmToCStrSafe(this->Makefile->GetDefinition("CMAKE_EXPORT_LINK_COMMANDS"));
```

Comment:
Could you please add option "LAST" so that cmake generates link_commands.json with the last available version

Reply:
CMAKE_EXPORT_LINK_COMMANDS=LAST

Reply:
CMAKE_EXPORTLINK_COMANDS=ON is better